test/sample-test/configs/lightweight_component.config.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+test_name: lightweight_component
+notebook_params:
```

Comment:
I think we can have the same way of configuring arguments for both notebooks and pipelines.
BTW, we can also pass data to notebooks by using the environment variables (`EXPERIMENT_NAME = os.environ['EXPERIMENT_NAME']`). This way the user can set the environment variable once and then run samples without modifications.

Reply:
The reason for the current implementation is that, these two sets of parameters are consumed by different things (one is papermill and another is kfp pipeline), and actually it's possible to have duplicate names across this two sets. Also, there functions are a bit of different.

Just want to make sure I under stand your second point correctly. Does it mean that we will encourage users to assign values in their notebook in this way? This config file will be used by the sample test infra only.

Reply:
>these two sets of parameters are consumed by different things (one is papermill and another is kfp pipeline)

Does papermill consume them itself or does it pass them to the notebook?

>Just want to make sure I under stand your second point correctly. 

The second point was about different ways the user or tests can set the required variables in a notebook. At this moment the user must manually insert variable values in a notebook and tests use papermill arguments. Maybe in future we can:
1. Reduce the number of variables that the user needs to specify to run the sample notebook (ideally to 0)
2. Maybe we can use environment variables for the rest. Papermill way of passing variables requires the sample notebooks to have specific structure which is not always the most readable (for example, previously all the component images had to be taken out of their components and specified in the first cell). This scheme would also be compatible with notebooks converted to python files.

I'm not asking you to implementing this. This is more like design thoughts that you might find relevant.

Reply:
> Does papermill consume them itself or does it pass them to the notebook?
papermill passes them to the notebook to substitute some constants defined in the notebook.

> Reduce the number of variables that the user needs to specify to run the sample notebook (ideally to 0)
Agree this would be ideal.

> Maybe we can use environment variables for the rest.
Does this require that something needs to be assigned using os.environ inside the notebook?

Reply:
> Does this require that something needs to be assigned using os.environ inside the notebook?

Yes. But this is something for another PR.

Reply:
I think the explicit way of putting the argument at the top is in fact a good practice. Depending on the environment variable for input is implicit and might lead to negligence.